batch/batch/driver/instance_pool.py

```diff
@@ -152,46 +166,100 @@ async def configure(
     def n_instances(self):
         return len(self.name_instance)
 
-    def adjust_for_remove_instance(self, instance):
-        assert instance in self.instances_by_last_updated
+    def add(self, instance):
+        assert instance.name not in self.name_instance
+        assert instance not in self.instances_by_last_updated
+        assert instance.state == 'pending'
 
-        self.instances_by_last_updated.remove(instance)
+        self.name_instance[instance.name] = instance
+        self.n_instances_by_state[instance.state] += 1
+        self.instances_by_last_updated.add(instance)
+        if instance.state in ('pending', 'active'):
+            self.live_free_cores_mcpu += max(0, instance.free_cores_mcpu)
+            self.live_total_cores_mcpu += instance.cores_mcpu
+        if instance.is_healthy():
+            self.healthy_instances_by_free_cores.add(instance)
 
-        self.n_instances_by_state[instance.state] -= 1
+    # async def create_instance should be here but it is huge currently
```

Comment:
You can't do this here. You call `add` every time you add an instance such as on startup.